crates/sequencer/src/state.rs

```diff
@@ -253,4 +312,75 @@ mod tests {
         // When
         state.get_compiled_class_hash(*ONE_CLASS_HASH).unwrap();
     }
+
+    #[test]
+    pub fn dump_and_load_state() {
+        let mut state = State::default();
+
+        // setting up entry for state.classes
+        let class_hash = ClassHash(stark_felt!("0x1"));
+        let contract_class = get_contract_class(include_str!(
+            "./test_data/contracts/compiled/universal_deployer.json"
+        ));
+        let compiled_class_hash = CompiledClassHash(stark_felt!("0x1"));
+        let contract_address = ContractAddress(patricia_key!("0x1"));
+        let contract_storage_key: ContractStorageKey =
+            (contract_address, StorageKey(patricia_key!("0x1")));
+        let storage_value = stark_felt!("0x1");
+        let nonce = Nonce(stark_felt!("0x1"));
+
+        state.classes.insert(class_hash, contract_class);
+        state
+            .compiled_class_hashes
+            .insert(class_hash, compiled_class_hash);
+        state.contracts.insert(contract_address, class_hash);
+        state.storage.insert(contract_storage_key, storage_value);
+        state.nonces.insert(contract_address, nonce);
+
+        let dump_file_path = PathBuf::from("./src/test_data/katana_dump.json");
+
+        state
+            .dump_state_to_file(&dump_file_path)
+            .unwrap_or_else(|error| {
+                panic!(
+                    "Failed to save state to path {:?},\n error {}",
+                    dump_file_path, error
+                )
+            });
+
+        let loaded_state = State::load_state_from_file(&dump_file_path).unwrap_or_else(|error| {
+            panic!(
+                "failed loading state from path {:?},\n error {}",
+                dump_file_path, error
+            )
+        });
+
+        assert_eq!(
+            state.classes.get(&class_hash),
+            loaded_state.classes.get(&class_hash)
+        );
+        assert_eq!(
+            state.compiled_class_hashes.get(&class_hash),
+            loaded_state.compiled_class_hashes.get(&class_hash)
+        );
+        assert_eq!(
+            state.contracts.get(&contract_address),
+            loaded_state.contracts.get(&contract_address)
+        );
+        assert_eq!(
+            state.storage.get(&contract_storage_key),
+            loaded_state.storage.get(&contract_storage_key)
+        );
+        assert_eq!(
+            state.nonces.get(&contract_address),
+            loaded_state.nonces.get(&contract_address)
+        );
```

Comment:
Can't we derive `PartialEq` and just have one `assert_eq!`?

Reply:
Great suggestion! Implemented via [this](https://github.com/kkrt-labs/ef-tests/pull/159/commits/87e3c996cdad92f3242f915f5666c559fb74b374) commit.